jazzhands/__main__.py

```diff
@@ -184,25 +209,57 @@ def main(argv=sys.argv):
     # collect, build, run or the setup command.
     # build implies collect, run implies build and collect. setup runs on its own.
 
-    if len(sys.argv) <= 1:
+    parser = argparse.ArgumentParser()
+    parser.set_defaults(which=None)
+    # parser.add_argument('command', type=str, action='store')
+    subparsers = parser.add_subparsers()
+    parsers = {}
+    def add_command(cmd):
+        parsers[cmd] = subparsers.add_parser(cmd)
+        parsers[cmd].set_defaults(which=cmd)
+
+    add_command('setup')
+    parsers['setup'].add_argument('-p', '--preset', action='append', type=str, default=[])
+    parsers['setup'].add_argument('-t', '--transform', action='append', type=str, default=[])
+
+    add_command('collect')
+
+    add_command('build')
+
+    add_command('run')
+    parsers['run'].add_argument('--auto-npm', action='store_true')
+
+    args = parser.parse_args()
+
```

Comment:
If I install this branch in voters-ingestor, and run jazzhands without arguments, it fails with a stack trace:

    $ jazzhands 
    Traceback (most recent call last):
      File "/home/dpoirier/.virtualenvs/voters-ingestor/bin/jazzhands", line 11, in <module>
        load_entry_point('jazzhands==0.1', 'console_scripts', 'jazzhands')()
      File "/home/dpoirier/.virtualenvs/voters-ingestor/lib/python3.6/site-packages/jazzhands/__main__.py", line 344, in main
        if index_files['js'] and js_dir:
    KeyError: 'js'

Reply:
I ran `jazzhands setup` apparently successfully in voters-ingestor, then `jazzhands collect` failed at the same place.

Reply:
I installed it in typecoach, cded to the webapp directory, and it ran without error without arguments, and did setup okay. Then `jazzhands collect` printed "Could not locate your project's main package. Jazzhands tries to locate this relative to a `settings` package or `settings.py` module."  There's a settings.py file in that directory.

Reply:
It is run from the top of the repository and will locate everything. It has to be at the top to find everything relative to the project's root folder.

Reply:
I also don't consider the failure on voters-ingestor to be a problem, because that project has no frontend to build so Jazzhands is not applicable.

Reply:
In typecoach, the whole Django app is under webapp, including manage.py and settings. I don't see why I shouldn't expect jazzhands to be able to find everything since it's all there.

Reply:
The Django project is all under `webapp` but Jazzhands is concerned with more than just the Django project. The `.babelrc` configuration and `node_modules` needed for the frontend building, which is its primary purpose, need to be at the repository level where those things live.

Reply:
In that project, ``node_modules`` actually is under ``webapp``.  

In any case, it seems perfectly reasonable for a Django project to have a settings.py in its top-level directory. Is there some reason to disallow that?